<a href="https://colab.research.google.com/github/Jacky-gif310/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [42]:
!git clone https://github.com/flyrank-bih/flyrank-ml-internship-starter.git

Cloning into 'flyrank-ml-internship-starter'...
remote: Enumerating objects: 175, done.
remote: Counting objects: 100% (43/43), done.
remote: Compressing objects: 100% (19/19), done.
remote: Total 175 (delta 33), reused 24 (delta 24), pack-reused 132 (from 2)
Receiving objects: 100% (175/175), 1.84 MiB | 8.57 MiB/s, done.
Resolving deltas: 100% (68/68), done.


In [43]:
!ls

AGENTS.md    docs			    notebooks	      scripts	  work
CLAUDE.md    flyrank-ml-internship-starter  outputs	      SETUP.md
data	     GUIDE.md			    README.md	      skills
DATA_USE.md  LICENSE			    requirements.txt  submission


In [44]:
!ls flyrank-ml-internship-starter

AGENTS.md  DATA_USE.md	LICENSE    README.md	     SETUP.md	 work
CLAUDE.md  docs		notebooks  requirements.txt  skills
data	   GUIDE.md	outputs    scripts	     submission


In [45]:
import os

os.getcwd()

'/content/flyrank-ml-internship-starter/flyrank-ml-internship-starter'

In [46]:
os.listdir()

['SETUP.md',
 'CLAUDE.md',
 'LICENSE',
 'notebooks',
 '.git',
 'DATA_USE.md',
 'AGENTS.md',
 'data',
 'requirements.txt',
 '.github',
 'flyrank-ml-internship-starter',
 'submission',
 'GUIDE.md',
 'README.md',
 'work',
 'docs',
 'skills',
 'outputs',
 '.gitignore',
 'scripts']

In [47]:
%cd flyrank-ml-internship-starter

/content/flyrank-ml-internship-starter/flyrank-ml-internship-starter/flyrank-ml-internship-starter


In [48]:
import os
os.listdir()

['SETUP.md',
 'CLAUDE.md',
 'LICENSE',
 'notebooks',
 '.git',
 'DATA_USE.md',
 'AGENTS.md',
 'data',
 'requirements.txt',
 '.github',
 'submission',
 'GUIDE.md',
 'README.md',
 'work',
 'docs',
 'skills',
 'outputs',
 '.gitignore',
 'scripts']

In [49]:
import pandas as pd

df = pd.read_csv(
    "data/raw/content_refresh_anonymized.csv"
)

df.head()

,content_id,client_id,search_volume,competition,competition_level,cpc,content_type,main_intent,word_count,char_count,...,char_count_tier,ctr,avg_position,engagement_rate,scroll_rate,ai_traffic_pct,impression_tier,position_tier,trend_direction,trend_pct
0,content_304f48230142,client_f369cb89fc,10.0,0.67,HIGH,2.05,keyword article,transactional,3221.0,20457.0,...,15000-25000,0.76,10.6,5.88,4.55,0.0,good,striking,down,-41.4
1,content_a1fb4e703a9e,client_4e07408562,90.0,0.01,LOW,0.05,keyword article,informational,2481.0,15562.0,...,15000-25000,0.05,20.3,0.00,10.00,0.0,good,page_3_5,down,-57.7
2,content_9aa793d4d895,client_7f2253d7e2,0.0,0.00,LOW,0.00,keyword article,informational,3515.0,23643.0,...,15000-25000,0.09,36.5,0.00,28.57,0.0,good,page_3_5,down,-60.9
3,content_331d6c4de07b,client_19581e27de,10.0,0.00,LOW,0.00,keyword article,commercial,NaN,NaN,...,NaN,0.49,6.2,1.28,3.45,0.0,good,page_1,stable,-13.8
4,content_d99b7a2d90ca,client_3fdba35f04,0.0,0.00,LOW,0.00,keyword article,informational,2803.0,17469.0,...,15000-25000,0.13,44.0,0.00,24.29,0.0,good,page_3_5,down,-34.7


# My Lane: Refresh / Content Opportunity Scoring


Refresh / Content Opportunity Scoring

I selected this lane because my goal is to identify which content pages may deserve human review based on observable search performance signals.

The purpose is not to automatically decide which pages should be changed, but to help prioritize which pages a reviewer should examine first.

This lane interests me because businesses have many pages, but limited time. A machine learning approach can help prioritize which pages deserve human attention.

## Research Question

### Decision

Which pages should be reviewed first for possible content improvement?

### Action

A content team could use the ranking to decide which pages need investigation, improvement, monitoring, or protection.

### Cost of a Wrong Recommendation

A wrong recommendation could waste time and resources by improving pages that do not need changes, while important pages may be ignored.

### Why Machine Learning Can Help

Machine learning can identify patterns in search and engagement data that may not be obvious from simple rules.

In [50]:
# Number of rows and columns

df.shape

(30000, 44)

In [51]:
# Check trend direction distribution

df["trend_direction"].value_counts()

,count
trend_direction,
down,16262
stable,5962
up,4388
new,2236
flat,1152


In [52]:
# Average impressions and CTR

df[["search_volume","ctr","avg_position"]].describe()

,search_volume,ctr,avg_position
count,27532.000000,30000.000000,30000.00000
mean,158.882391,0.510733,16.34238
std,1518.270825,3.279162,15.21679
min,0.000000,0.000000,0.00000
25%,0.000000,0.000000,6.20000
50%,10.000000,0.070000,10.80000
75%,20.000000,0.290000,22.30000
max,74000.000000,100.000000,245.00000


## Quick Look at the Data

I explored the dataset to understand the available signals before choosing a machine learning approach.

The dataset contains 30,000 content records with search and engagement information.

Some observations:

- The dataset has 27,532 pages with available search volume data.
- The average search volume is approximately 158.88, showing that pages have different levels of search demand.
- The average CTR is approximately 0.51, which helps measure how often pages receive clicks from impressions.
- The average search position is approximately 16.34, meaning many pages are not ranking among the highest search positions.

These signals suggest that search performance metrics can help identify which pages may need further review or improvement.

## What I Can and Cannot Claim

Based on this analysis, I can identify patterns between content performance signals and pages that may need attention.

I cannot claim that changing a page will automatically improve rankings or traffic.

The data helps create recommendations for human review, but final decisions should consider additional context and expert judgment.

# Self Check

✓ I selected a predefined FlyRank lane.

✓ I defined a clear research question.

✓ I identified the decision this analysis supports.

✓ I described possible actions based on the output.

✓ I considered the cost of incorrect recommendations.

✓ I used real numbers from the starter dataset.

✓ I avoided making claims beyond what the data can prove.